Jenkinsfile.trigger

```
@@ -102,69 +102,100 @@ if (breakEarly) { return } // thanks Jenkins...
-									-H 'X-GitHub-Api-Version: 2022-11-28' \
-									https://api.github.com/repos/docker-library/meta/actions/workflows/build.yml/dispatches \
-									-d "$payload"
-							'''
+							withCredentials([
+								string(
+									variable: 'GH_TOKEN',
+									credentialsId: 'github-access-token-docker-library-bot-meta',
+								),
+							]) {
+								sh '''
+									set -u +x
+
+									# https://docs.github.com/en/free-pro-team@latest/rest/actions/workflows?apiVersion=2022-11-28#create-a-workflow-dispatch-event
+									curl -fL \
+										-X POST \
+										-H 'Accept: application/vnd.github+json' \
+										-H "Authorization: Bearer $GH_TOKEN" \
+										-H 'X-GitHub-Api-Version: 2022-11-28' \
+										https://api.github.com/repos/docker-library/meta/actions/workflows/build.yml/dispatches \
+										-d "$payload"
+								'''
+							}
 						}
+						// record that GHA was triggered (for tracking continued triggers that fail to push an image)
+						buildCompletionData[buildObj.buildId] = [
+							lastTime: System.currentTimeMillis() / 1000, // convert to seconds
+							url: currentBuild.absoluteUrl,
+						]
 					}
-					// record that GHA was triggered (for tracking continued triggers that fail to push an image)
-					buildCompletionData[buildObj.buildId] = [
-						lastTime: System.currentTimeMillis() / 1000, // convert to seconds
-						url: currentBuild.absoluteUrl,
-					]
 				}
-			} else {
+			}
+		} else {
+			// "catchError" to set "stageResult" :(
+			catchError(message: 'Build of "' + buildObj.identifier + '" failed', buildResult: 'UNSTABLE', stageResult: 'FAILURE') {
+
+				// why not parallel these build() invocations?
+				// jenkins parallel closures get started in a randomish order, ruining our sorted queue
 				def res = build(
 					job: 'build',
 					parameters: [
 						string(name: 'buildId', value: buildObj.buildId),
+						string(name: 'identifier', value: buildObj.identifier),
 					],
 					propagate: false,
-					quietPeriod: 5, // seconds
+					// trigger these quickly so they all get added to Jenkins queue in "queue" order (also using "waitForStart" means we have to wait for the entire "quietPeriod" before we get to move on and schedule more)
+					quietPeriod: 0, // seconds
+					// we'll wait on the builds in parallel after they are all queued (so our sorted order is the queue order)
+					waitForStart: true,
 				)
-				// record the job failure
-				buildCompletionData[buildObj.buildId] = [
-					lastTime: (res.startTimeInMillis + res.duration) / 1000, // convert to seconds
-					url: res.absoluteUrl,
-				]
-				if (res.result != 'SUCCESS') {
-					// set stage result via catchError
-					error(res.result)
-				}
+				waitQueue[buildObj.identifier] = waitQueueClosure(buildObj.identifier, buildObj.buildId, res.externalizableId)
 			}
 		}
 	}
 }
 
+// wait on all the 'build' jobs that were queued
+if (waitQueue.size() > 0) {
+	parallel waitQueue
```

I usually prefer to be explicit about the `()` in single-argument invocations like this (although I don't usually go as far as "naming" the parameter), but I don't think it really matters much either way. :+1:

```suggestion
	parallel(waitQueue)
```